mocha-snippets.el

```diff
@@ -100,5 +101,24 @@ E.g.
           (format  "(%s)%s=>" params space)
         (format "function%s(%s)" space params))))
 
+(defun mocha-snippets-ts-function-declaration (&optional isAsync)
+  "Function head appropriate for the desired syntax.
+The user can configure whether to use the ES6 function syntax or the 'classic'
+function syntax.  This will return the appropriate declaration depending on
+which is configured: either 'function(): void' or '(): void=>'.
+
+PARAMS, will be substituded as the parameter list for the function.
+E.g.
+
+  (mocha-snippets-initialize \"hello, world\") => function(hello, world)"
+  (let (
+        (beforeFunc (if isAsync "async " ""))
+        (returnType (if isAsync "Promise<void>" "void"))
+        (space (if mocha-snippets-add-space-after-function-keyword " " ""))
+       )
+      (if mocha-snippets-use-fat-arrows
+          (format  "%s()%s: %s=>" beforeFunc space returnType))
```

Comment:
Perhaps should give additional vars similar to `mocha-snippets-add-space-after-function-keyword` to control the TS specific types spacing..

` "%s()%s: %s=>" `  -> `"%s()%s:%s%s%s=>"  beforeFunc space spaceAfterTypeColon returnType spaceAfterReturnType`

Reply:
Yeah, I've been thinking how awesome it would be if the configuration were read from es/ts lint